warehouse/ingest-core/src/main/java/datawave/ingest/util/cache/JobCacheFactory.java

```diff
@@ -0,0 +1,195 @@
+     *            An object that contains the job cache base path and its associated file system.
+     * @return A collection objects representing the status for each job cache.
+     */
+    public Collection<JobCacheStatus> getJobCacheStatuses(FileSystemPath baseCachePath) {
+        // @formatter:off
+        return getCacheCandidates(baseCachePath, timestampPattern, 0)
+                .stream()
+                .map(this::getJobCacheStatus)
+                .collect(Collectors.toList());
+        // @formatter:on
+    }
+    
+    /**
+     * Will attempt to release all locks for each cache id stored.
+     */
+    public void close() {
+        jobIdToCacheId.clear();
+        lockFactory.close();
+    }
+    
+    /**
+     * Find job cache paths that pass the input criteria.
+     *
+     * @param cacheBasePath
+     *            An object that contains the job cache base path and its associated file system.
+     * @param pattern
+     *            A pattern that will be used to filter job cache paths.
+     * @param keepNumVersions
+     *            The number of versions to skip(this is used primarily when deleting)
+     * @return A collection of job cache paths sorted in reverse order.
+     */
+    public static Collection<FileSystemPath> getCacheCandidates(FileSystemPath cacheBasePath, Pattern pattern, int keepNumVersions) {
+        Path cachePath = cacheBasePath.getOutputPath();
+        FileSystem fileSystem = cacheBasePath.getFileSystem();
+        
+        FileStatus[] pathStatus = FilesFinder.listPath(fileSystem, cachePath, pattern);
+        
+        // @formatter:off
+        return Arrays
+                .stream(pathStatus)
+                .filter(FileStatus::isDirectory)
+                .map(FileStatus::getPath)
+                .sorted(PATH_COMPARATOR.reversed())
+                .skip(keepNumVersions)
+                .map(deleteCachePath -> new FileSystemPath(fileSystem, deleteCachePath))
+                .collect(Collectors.toList());
+        // @formatter:on
+    }
+    
+    JobCacheLockFactory getLockFactory() {
+        return lockFactory;
+    }
+    
+    /**
+     * Find the job cache status for this job cache path
+     *
+     * @param cachePath
+     *            An object that contains the job cache path and its associated file system.
+     * @return An object that represents the status of the job cache.
+     */
+    private JobCacheStatus getJobCacheStatus(FileSystemPath cachePath) {
+        String lockIdPath = getLockIdPath(cachePath.getOutputPath().getName());
+        return new JobCacheStatus(cachePath, lockFactory.getCacheStatus(lockIdPath));
+    }
+    
+    /**
+     * Form the lock id by joining the passed in array with file.separator as the delimiter.
+     *
+     * @param lockIdElements
+     *            An array of lock id components that when joined form the lock id.
+     * @return A String representing the lock id.
+     */
+    String getLockIdPath(String... lockIdElements) {
```

Comment:
This method is really generic. It could even go in `FileUtils`. Maybe a `FileUtils.toPath(String... elements)` or something like that?

Reply:
I agree that it is generic but the FileUtils is under the commons-test.  I am not sure if I want to create a FileUtils under commons for one method.  I am not sure if it is a good fit under the Files class.  Ok to leave it here?

Reply:
Ahh, sure - that's cool. We could rename the method since it's not necessarily specific to forming a LockId path but more just creating a file path from the pieces.